code/datums/components/crafting/recipes.dm

```diff
@@ -142,6 +142,142 @@
 	category = CAT_WEAPONRY
 	subcategory = CAT_WEAPON
 
+/datum/crafting_recipe/advancedegun
+	name = "Advanced Energy Gun"
+	tools = list(TOOL_SCREWDRIVER, TOOL_MULTITOOL)
+	result = /obj/item/gun/energy/e_gun/nuclear
+	reqs = list(/obj/item/gun/energy/e_gun = 1,
```

Comment:
The fact that this creates guns with a firing pin when there could be no firing pin in the original gun is a bit weird. Can you add an override to check_requirements that checks if a pin is installed?

Reply:
Honestly, I wonder if it'd be more functionally better to find a method to have the gun spawned not have a pin if the base gun didn't...

Reply:
it would be but thatll be harder.